extensions/amp-analytics/0.1/events.js

```diff
@@ -389,6 +425,42 @@ export class VisibilityTracker extends EventTracker {
   }
 
   /**
+   * @param {string|undefined} waitForSpec
+   * @param {string|undefined} selector
+   * @param {?Element} element
+   * @return {?Promise}
+   * @visibleForTesting
+   */
+  getReadyPromise(waitForSpec, selector, element) {
+    if (!selector) {
+      // Default case #1: wait for nothing if selector is not specified.
+      return null;
```

Comment:
I agree. This is really the backward compatibility issue since the event was defined originally as `selector=undefined` and `waitFor=none`. That being said, I think `waitFor` creates a positive change and clarifies it sufficiently. Currently we basically have the same state, but not sufficiently clarified or customizable.